### ISSUE TITLE
Remote debugging: Could not find file descriptor referring to console

### ISSUE THREAD

**Issue by antoniusostermann** [teampoltergeist/poltergeist#715]
If I want to start a remote debugging session with `page.driver.debug`, I get this message in my console: `Could not find file descriptor referring to console`. The browser does not open. 

It seems to be a very old issue since there is an unanswered comment in this blog entry http://www.jonathanleighton.com/articles/2012/poltergeist-0-6-0/ describing the same issue.

I am using Kubuntu with Linux

```
Linux version 4.2.0-19-generic (buildd@lgw01-28) (gcc version 5.2.1 20151010 (Ubuntu 5.2.1-22ubuntu2) ) #23-Ubuntu SMP Wed Nov 11 11:39:30 UTC 2015
```

and these relevant Gems:

```
Using websocket-extensions 0.1.2
Using websocket-driver 0.6.3
Using poltergeist 1.8.1
Using capybara 2.5.0
Using capybara-ember 0.0.1
Using launchy 2.4.3
```

If a open my browser on my own and point it to `https://localhost:9494` (or `http://localhost:9494`, my page uses SSL though), it can't establish a connection / says that no server is running.

Anyone may help me? :-)

**Comment by twalpole** [comment]
What config are you using for the driver? What version of PhantomJS?

**Comment by Canifex** [comment]
Sorry for the late reply.

My driver config:
```ruby
Capybara.register_driver :poltergeist do |app|
  Capybara::Poltergeist::Driver.new(app, js_errors: false, inspector: 'open')
end
Capybara.javascript_driver = :poltergeist
```

My installed phantom js version is 1.9.8.

**Comment by twalpole** [comment]
Please try `inspector: true` and see if the result is any different

**Comment by Canifex** [comment]
Tried this, but same result

**Comment by route** [comment]
This is really weird. Does it raise en error? If so can you show us backtrace?

**Comment by Canifex** [comment]
Nope, it doesn't.

**Comment by schnittchen** [comment]
The location of the issue is in Inspector#open:

```
[1] pry(#<Capybara::Poltergeist::Inspector>)> Process.spawn(browser, url) 
Couldn't get a file descriptor referring to the console
=> 21880
```

It seems to work with `xdg-open` instead of `open`, though I'm using the feature for the first time and don't know what I really can expect.

**Comment by tisba** [comment]
I'm running into the same issue. This is running in a headless VM, so there is no open or `xdg-open` or alike. Could that be the problem in my case?

```
$ uname -a
Linux applejack 3.13.0-100-generic #147-Ubuntu SMP Tue Oct 18 16:48:51 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

Configuration:

```
Capybara.register_driver :poltergeist_debug do |app|
  Capybara::Poltergeist::Driver.new(app, {
    js_errors: true,
    phantomjs_options: %w(--proxy-type=none),
    timeout: 180,
    inspector: true
  })
end
Capybara.javascript_driver = :poltergeist_debug
```

Output:

```
All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 13985
|  ETA: ??:??:??
Poltergeist execution paused. Press enter (or run 'kill -CONT 29755') to continue.
Couldn't get a file descriptor referring to the console
```

**Comment by twalpole** [comment]
@tisba Yes if there isn't a way to open a browser then it can't open a browser

**Comment by tisba** [comment]
let me rephrase: what about a more useful error message and, more important: print the debug URL.

**Comment by twalpole** [comment]
@tisba The problem is we don't know what the debug URL is.  PhantomJS binds its remote debugger to all available interfaces on port 9664 .  We have no way of knowing which interface a user would need to connect to (Poltergeist tries to launch a browser to connect to localhost:9664 but that's not much use to a user if the machine poltergeist is running on can't launch a browser) from the machine they want to run their own browser on.  
Since this appears to be a user issue I'm closing this.  PRs that clarify the error message or pause to allow a user to connect from a different machine if wanted will be considered.
